main.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/NickGowdy/deveui-cli/client"
+	"github.com/NickGowdy/deveui-cli/device"
+	"github.com/NickGowdy/deveui-cli/processor"
+	"github.com/joho/godotenv"
+)
+
+/*
+deveui-cli is a Go CLI program.
+It is used for concurrently generating unique 16-character (hex) identifier called a DevEUI.
+These are generated by the program and registered via an external (LoRaWAN) API.
+
+Usage:
+
+	go run main.go (locally)
+
+	go run deveui-cli (docker)
+
+Once this program starts, it will listen to syscall.SIGTERM, syscall.SIGINT via a channel.
+This is to handle any unexpected terminations of the program and to resume processing DevEUIs.
+*/
+func main() {
+	if err := godotenv.Load(".env"); err != nil {
+		panic("error loading.env file")
+	}
+	godotenv.Load(".env")
+	baseurl := os.Getenv("BASE_URL")
+
+	maxConcurrentJobs, err := strconv.Atoi(os.Getenv("MAX_CONCURRENT_JOBS"))
+	if err != nil {
+		panic("error parsing MAX_CONCURRENT_JOBS to int")
+	}
+
+	codeRegistrationLimit, err := strconv.Atoi(os.Getenv("CODE_REGISTRATION_LIMIT"))
+	if err != nil {
+		panic("error parsing CODE_REGISTRATION_LIMIT to int")
+	}
+
+	timeout, err := strconv.Atoi(os.Getenv("TIMEOUT"))
+	if err != nil {
+		panic("error parsing TIMEOUT to int")
+	}
+
+	// setup client for requests
+	httpClient := &http.Client{
+		Timeout: time.Second * time.Duration(timeout),
+	}
+
+	loraWAN := client.NewLoraWAN(baseurl, httpClient)
+
+	// setup processor to do work
+	codeProcessor := &processor.CodeProcessor{
+		CodeRegistrationLimit: 1,
+		MaxConcurrentJobs:     maxConcurrentJobs,
+		LoraWAN:               *loraWAN,
+		DeviceCh:              make(chan device.Device, maxConcurrentJobs),
+		DoneCh:                make(chan struct{}),
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	workCh := make(chan device.Device, maxConcurrentJobs)
+	doneCh := make(chan struct{})
+	count := 0
+
+	// Fill work buffer so we can start processing work
+	go func(count int) {
+		for count < 100 {
+			workCh <- device.Device{}
+		}
+		return
+	}(count)
+
+	// Spawn workers
+	for job := 0; job < codeRegistrationLimit; job++ {
```

Comment:
It would be neater if `CodeProcessor` had a `.Run()` method that span up the worker GoRoutines. This method could accept a `context.Context` to handle shutdown, or `CodeProcessor` could have a method called `Stop()`.